torchvision/models/detection/generalized_rcnn.py

```diff
@@ -36,19 +36,26 @@ def __init__(self, backbone, rpn, roi_heads, transform):
         self._has_warned = False
 
     @torch.jit.unused
-    def eager_outputs(self, losses, detections):
-        # type: (Dict[str, Tensor], List[Dict[str, Tensor]]) -> Union[Dict[str, Tensor], List[Dict[str, Tensor]]]
+    def eager_outputs(
+        self,
+        losses: Dict[str, Tensor],
+        detections: List[Dict[str, Tensor]],
+    ) -> Union[Dict[str, Tensor], List[Dict[str, Tensor]]]:
+
         if self.training:
             return losses
 
         return detections
 
-    def forward(self, images, targets=None):
-        # type: (List[Tensor], Optional[List[Dict[str, Tensor]]]) -> Tuple[Dict[str, Tensor], List[Dict[str, Tensor]]]
+    def forward(
+        self,
+        images: List[Tensor],
+        targets: Optional[List[Dict[str, Tensor]]] = None,
+    ) -> Union[Tuple[Dict[str, Tensor], List[Dict[str, Tensor]]], Dict[str, Tensor], List[Dict[str, Tensor]]]:
```

Comment:
@datumbox This change causes failure in internal tests, check D32216673.

Reply:
FYI
This change was done as Union is supported by Torchscript. There isn't a CI failure on main branch. (Probably you are referring to fbcode)

Reply:
Yeah.
```
RuntimeError: 
Union[Tuple[Dict[str, Tensor], List[Dict[str, Tensor]]], List[Dict[str, Tensor]], Dict[str, Tensor]] cannot be used as a tuple:
  File "/data/sandcastle/boxes/eden-trunk-hg-fbcode-fbsource/fbcode/buck-out/dev/gen/smart/smart_catalog_filter/inference/handler_test#binary,link-tree/smart_catalog_filters/embedding.py", line 144
    def forward(self, images: List[Tensor]):
        _, results = self.rcnn(images)
                     ~~~~~~~~~~~~~~~~ <--- HERE
        return [
            (self.get_emb(result["features"]), result["boxes"]) for result in results


----------------------------------------------------------------------
```

Reply:
@prabhat00155 Thanks for flagging, I see it. This is something we should keep in mind for future Annotations and PRs. 

Union of a tuple with a non-tuple is likely to break code because of expressions like:
`a, b = result` 

My proposal is to amend the diff on FBcode to either remove the annotation or remove the union (whichever passes all tests) and then sync the change that you will do on FBcode to GH on a cherrypick OR on a new PR (Which ever easier). 

Let me know your thoughts on the above and if I can do something to help.

Reply:
Thanks, I'll amend the diff.